client/src/components/SearchPage/SearchField/SearchField.js

```diff
@@ -79,21 +82,33 @@ class SearchField extends Component {
                 </div>
                 <div className={classes.search}>
                     <div>
-                        <img src={Search} className={classes.srch} />
+                        <img
+                            src={Search}
+                            className={classes.srch}
+                            onClick={() => this.setState({ redirect: true })}
+                        />
                         <input
-                            type="text"
+                            type='text'
                             value={this.state.userInput}
-                            onChange={(e) => this.handleUserInput(e)}
+                            onInput={(e) => this.handleUserInput(e)}
                             onKeyDown={(e) => this.handleKeyPress(e)}
+                            onChange={(e) =>
+                                this.setState({ query: e.target.value })
+                            }
                         />
+                        {this.state.redirect ? (
+                            <Redirect
```

Comment:
use Link instead of Redirect (here as well as at all the other occurrences in this file). That will enable the user to click the back button to go back to the search screen